### ISSUE TITLE
Permission denied after upgrade to 3.0

### ISSUE THREAD

**Issue by viaregio** [Kozea/Radicale#1047]
After I upgraded my installation from version 2.1.12 with

```
pip install -U radicale
```

I can no longer login.

My config file contains:

```
[auth]
type = http_x_remote_user
```

I don't use a rights file like in issue #1044 

Output of logging:

```
ai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [INFO] PROPFIND request for '/principals/' with depth '0' received from 127.0.0.1 using 'Mac+OS+X/10.11.6 (15G22>
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [DEBUG] Request headers:
Mai 22 10:25:14 lanai radicale[23371]: {'CONTENT_LENGTH': '170',
Mai 22 10:25:14 lanai radicale[23371]:  'CONTENT_TYPE': 'text/xml',
Mai 22 10:25:14 lanai radicale[23371]:  'GATEWAY_INTERFACE': 'CGI/1.1',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_ACCEPT': '*/*',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_ACCEPT_ENCODING': 'gzip, deflate',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_ACCEPT_LANGUAGE': 'de-de',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_AUTHORIZATION': 'Basic **masked**',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_BRIEF': 't',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_CONNECTION': 'close',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_DEPTH': '0',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_HOST': 'radicale.removed-domainname.de',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_PREFER': 'return=minimal',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_USER_AGENT': 'Mac+OS+X/10.11.6 (15G22010) CalendarAgent/361.2',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_X_FORWARDED_PROTO': 'https',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_X_FORWARD_FOR': '192.168.2.3',
Mai 22 10:25:14 lanai radicale[23371]:  'HTTP_X_REMOTE_USER': 'familie',
Mai 22 10:25:14 lanai radicale[23371]:  'PATH_INFO': '/principals/',
Mai 22 10:25:14 lanai radicale[23371]:  'QUERY_STRING': '',
Mai 22 10:25:14 lanai radicale[23371]:  'REMOTE_ADDR': '127.0.0.1',
Mai 22 10:25:14 lanai radicale[23371]:  'REMOTE_HOST': '',
Mai 22 10:25:14 lanai radicale[23371]:  'REQUEST_METHOD': 'PROPFIND',
Mai 22 10:25:14 lanai radicale[23371]:  'SCRIPT_NAME': '',
Mai 22 10:25:14 lanai radicale[23371]:  'SERVER_NAME': 'localhost',
Mai 22 10:25:14 lanai radicale[23371]:  'SERVER_PORT': '5232',
Mai 22 10:25:14 lanai radicale[23371]:  'SERVER_PROTOCOL': 'HTTP/1.0',
Mai 22 10:25:14 lanai radicale[23371]:  'SERVER_SOFTWARE': 'WSGIServer/0.2',
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.errors': <_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>,
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.input': <_io.BufferedReader name=7>,
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.multiprocess': False,
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.multithread': True,
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.run_once': False,
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.url_scheme': 'http',
Mai 22 10:25:14 lanai radicale[23371]:  'wsgi.version': (1, 0)}
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [DEBUG] Sanitized script name: ''
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [DEBUG] Sanitized path: '/principals/'
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [INFO] Successful login: 'familie'
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [INFO] Access to '/principals/' denied for 'familie'
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [DEBUG] Response content:
Mai 22 10:25:14 lanai radicale[23371]: Access to the requested resource forbidden.
Mai 22 10:25:14 lanai radicale[23371]: [2020-05-22 10:25:14 +0200] [23371/Thread-36] [INFO] PROPFIND response status for '/principals/' with depth '0' in 0.001 seconds: 403 Forbidden
```

After downgrading to version 2.1.12 the error is gone and radicale starts working again.

**Comment by paolostivanin** [comment]
Same here! After upgrading to 3.0 I can no longer use DAVdroid.
What's weird though is that from the web ui and from a small python script everything works fine.
But from DAVdroid and Thunderbird no luck

**Comment by Unrud** [comment]
@paolostivanin Try version 3.0.1, maybe it is caused by #825.

@viaregio The client tries to access ``/principals/`` which gets denied. Version 2.1.12 should also deny access. I don't think that the problem is caused by this. You have to check the log for other problems. What client are you using?

**Comment by paolostivanin** [comment]
@Unrud I still get an HTTP 500 from davdroid:

```
May 22 17:24:43 samaritan radicale[10011]: [2020-05-22 17:24:43 +0200] [10011/Thread-1] [ERROR] An exception occurred during PROPFIND request on '/mycal/myuid/': [Errno 13] Permission denied: '/var/lib/radicale/collections/collection-root/mycal/myuid/.Radicale.cache/.Radicale.lock.item'

```

**Comment by viaregio** [comment]
After upgrading to version 3.0.1 I tried the following clients to login:

  - macOS calendar
  - iOS calendar
  - Python caldav package
  - The Web UI of radicale

None of the methods above succeded.

With version 2.1.12 all four methods succeed.

The log from the access with the Web UI contains:

```
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [INFO] PROPFIND request for '/' received from 127.0.0.1 using 'Mozilla/5.0 (Ma>
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [DEBUG] Request headers:
Mai 22 17:51:11 lanai radicale[212130]: {'CONTENT_LENGTH': '127',
Mai 22 17:51:11 lanai radicale[212130]:  'CONTENT_TYPE': 'text/plain;charset=UTF-8',
Mai 22 17:51:11 lanai radicale[212130]:  'GATEWAY_INTERFACE': 'CGI/1.1',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_ACCEPT': '*/*',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_ACCEPT_ENCODING': 'gzip, deflate, br',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_ACCEPT_LANGUAGE': 'de,en;q=0.7,en-US;q=0.3',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_AUTHORIZATION': 'Basic **masked**',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_CONNECTION': 'close',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_COOKIE': '**masked**',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_DNT': '1',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_HOST': 'radicale.domainname.de',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_ORIGIN': 'https://radicale.domainname.de',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_REFERER': 'https://radicale.domainname.de/.web/',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_USER_AGENT': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:76.0) '
Mai 22 17:51:11 lanai radicale[212130]:                     'Gecko/20100101 Firefox/76.0',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_X_FORWARD_FOR': '192.168.2.77',
Mai 22 17:51:11 lanai radicale[212130]:  'HTTP_X_REMOTE_USER': 'familie',
Mai 22 17:51:11 lanai radicale[212130]:  'PATH_INFO': '/',
Mai 22 17:51:11 lanai radicale[212130]:  'QUERY_STRING': '',
Mai 22 17:51:11 lanai radicale[212130]:  'REMOTE_ADDR': '127.0.0.1',
Mai 22 17:51:11 lanai radicale[212130]:  'REMOTE_HOST': '',
Mai 22 17:51:11 lanai radicale[212130]:  'REQUEST_METHOD': 'PROPFIND',
Mai 22 17:51:11 lanai radicale[212130]:  'SCRIPT_NAME': '',
Mai 22 17:51:11 lanai radicale[212130]:  'SERVER_NAME': 'localhost',
Mai 22 17:51:11 lanai radicale[212130]:  'SERVER_PORT': '5232',
Mai 22 17:51:11 lanai radicale[212130]:  'SERVER_PROTOCOL': 'HTTP/1.0',
Mai 22 17:51:11 lanai radicale[212130]:  'SERVER_SOFTWARE': 'WSGIServer/0.2',
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.errors': <_io.TextIOWrapper name='<stderr>' mode='w' encoding='utf-8'>,
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.file_wrapper': <class 'wsgiref.util.FileWrapper'>,
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.input': <_io.BufferedReader name=7>,
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.multiprocess': False,
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.multithread': True,
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.run_once': False,
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.url_scheme': 'http',
Mai 22 17:51:11 lanai radicale[212130]:  'wsgi.version': (1, 0)}
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [DEBUG] Sanitized script name: ''
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [DEBUG] Sanitized path: '/'
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [INFO] Successful login: 'familie'
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [DEBUG] Request content:
Mai 22 17:51:11 lanai radicale[212130]: <?xml version="1.0"?>
Mai 22 17:51:11 lanai radicale[212130]: <propfind xmlns="DAV:">
Mai 22 17:51:11 lanai radicale[212130]:   <prop>
Mai 22 17:51:11 lanai radicale[212130]:     <current-user-principal />
Mai 22 17:51:11 lanai radicale[212130]:     <displayname />
Mai 22 17:51:11 lanai radicale[212130]:   </prop>
Mai 22 17:51:11 lanai radicale[212130]: </propfind>
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [INFO] Access to '/' denied for 'familie'
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [DEBUG] Response content:
Mai 22 17:51:11 lanai radicale[212130]: Access to the requested resource forbidden.
Mai 22 17:51:11 lanai radicale[212130]: [2020-05-22 17:51:11 +0200] [212130/Thread-5] [INFO] PROPFIND response status for '/' in 0.002 seconds: 403 Forbidden
```

**Comment by Unrud** [comment]
> ```[INFO] Access to '/' denied for 'familie'```

What ``rights`` type are you using? I can't reproduce the problem and that error should only be possible with a rights file or a custom rights plugin.

> ```  'HTTP_X_FORWARD_FOR': '192.168.2.77',```

This is unrelated to the problem but the reverse proxy configuration contains a typo. It should be ``X-Forwarded-For`` not ``X-Forward-For``.

**Comment by viaregio** [comment]
My config file only contains the following text:

```
[auth]
type = http_x_remote_user
```

For debugging I added in version 3.0 and 3.0.1:

```
[logging]
level = debug
```

I have no rights configured.

**Comment by Unrud** [comment]
> I have no rights configured.

Maybe it fails because your root collection is a calendar or address book. Try to run ``radicale --verify-storage``.

**Comment by viaregio** [comment]
Yes, this has been the issue. Together with your tip from https://github.com/Kozea/Radicale/issues/977#issuecomment-575648203 this fixed it for me.

**Comment by bhartshorn** [comment]
I realize this issue is closed, but wanted to drop a solution for a similar problem. I was also received an `[Errno 13] Permission denied` error, but only when trying to sync using DAVx5.

I added a debug log to the code at the line it was failing to get the filepath that Radicale was struggling with. Turned out to be a file in one of the `.Radicale.cache/sync-token/` directories. I checked the permissions on the directories, which were fine (`750`), but it turned out that most of the `Radicale.cache` directories were owned by root:root. After a recursive chown of `/var/lib/radicale/` to radicale:radicale, everything is working again.

I'm not sure how the directories came under ownership of root. This is on an Arch Linux server using the official package and systemd unit. I had just upgraded to 3.0.x from some 2.x release.
